### PR TITLE
update Electron to 1.6.11

### LINKED ISSUE #671: Enterprise self-signed/untrusted certificate support
Refs #667 

It's not uncommon for GitHub Enterprise instances to run with a self-signed certificate or certificate signed by a non-authorative CA. There's several reasons why that might be true. These include:

 * a freshly launched GHE instance like a trial
 * of our AWS/Azure templates which generate a self-signed certificate on launch
 * GHE instances which run on non TLD domains (like ghe.lan) for which no CA will give out a certificate

We need to make sure that our enterprise sign in flow can handle an untrusted certificate and present enough information that the user can make an informed decision about whether to trust it permanently. I don't know how this works on Mac, maybe it's already taken care by the system, but on Windows we're going to have to do a bit of work for this to work nicely.

First, we need a way of installing a certificate into the appropriate certificate store. Second, we need a way of keeping the curl ca-bundle file in sync with the certificate store. This all works in the current version on Desktop on Windows. The implementation for the order of certificates in the ca-bundle is rather finicky but luckily we have a reference implementation in [MsysGitCertificateInstaller.cs](https://github.com/github/Windows/blob/master/GitHub/Helpers/MsysGitCertificateInstaller.cs) as well as a starting point for installing certificates in the store ([SystemCertificateInstaller.cs](https://github.com/github/Windows/blob/master/GitHub/Helpers/SystemCertificateInstaller.cs)). I'm assuming this will require us to write some native helpers on Windows.

This is the current flow on Windows

### PR DESCRIPTION
**Pull request by shiftkey**
So I was able to workaround much of the incompatibilities between the two type declarations, with some rough edges called out. It'd be nice to ensure the rough edges are sorted before merging, rather than letting it hang around.

@desktop/electron where's the best place to report these issues upstream - an open issue on `electron-typescript-declarations` walking through the samples and cross-referencing the API docs that need to be updated so it all flows through? Or something else?

Fixes #671 

TODO:

 - [x] y u no build
 - [x] verify self-signed certificate flow on macOS
 - [x] verify self-signed certificate flow on Windows

### CODE REVIEW

**Comment by kevinsawicki** [comment]
> This new version should be released today or tomorrow. Will keep you posted.

1.6.10 is now out and the only changes from 1.6.9 are doc fixes for the typescript definition file.

**Comment by shiftkey** [comment]
@zeke @kevinsawicki I opened https://github.com/electron/electron-typescript-definitions/issues/54 to track a case that wasn't supported with the old definitions that I'd hoped was fixed. But we can workaround that for now rather than wait another week.

**Comment by shiftkey** [comment]
Looks like something affects the unit tests task (this happens after the tests have all passed, blocking the unit tests from completing and thus the integration tests):

**Comment by shiftkey** [comment]
Blocked on https://github.com/electron/electron/issues/9583

**Comment by joshaber** [comment]
I opened a PR on electron-mocha that should fix this: https://github.com/jprichardson/electron-mocha/pull/105

**Comment by joshaber** [comment]
After updating `electron-mocha`, CI is passing on Appveyor: https://github.com/desktop/desktop/pull/1996

**Comment by shiftkey** [comment]
So close to passing CI for both platforms, this issue has returned and I'm trying to remember how we fixed this last time:

File: app/src/models/app-menu.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ export interface IMenu {
  */
 function getAccelerator(menuItem: Electron.MenuItem): string | null {
   if (menuItem.accelerator) {
-    return menuItem.accelerator
+    return menuItem.accelerator as string
```
Comment (shiftkey) [inline app/src/models/app-menu.ts]: For reference:

```
class Accelerator extends String {

}
```

**Comment by shiftkey** [comment]
Ready for review 🔴

**Comment by zeke** [comment]
🎉

**Comment by zeke** [comment]
@shiftkey can you distill this upgrade experience into a few words? Wondering if there's anything the Electron team can do to make these migrations easier for others.

**Comment by shiftkey** [comment]
@zeke most of the pain was around the differences in type declarations. A resource for migrating from the existing `@types/electron` would probably help, even though the list of breaking changes is small currently:

 - `IpcRendererEvent` and `IpcMainEvent` -> `IpcMessageEvent`
 - `MenuItemOptions` -> `MenuItemConstructorOptions`

There's still [these customizations](https://github.com/desktop/desktop/blob/b819b80d7eaa82ecba4c5bd96a30dec5af3e72c3/app/src/lib/globals.d.ts#L110-L143) that should live upstream somewhere (whether in `electron` or in `@types/node`), as you shouldn't need to carry these around between projects.

**Pull request merged**